model/base_constraints.py

```diff
@@ -135,10 +136,11 @@ def add_no_demand_cover_during_off_shift(self, w, x, y, version):
     def add_no_demand_cover_during_off_shift_original(self, w, y):
         self.model.addConstrs(
             (
-                len(self.t_in_off_shifts[t, v]) * w[e, t, v]
-                <= quicksum(
-                    quicksum((1 - y[c, e, t_mark]) for c in self.competencies)
-                    for t_mark in self.t_in_off_shifts[t, v]
+                quicksum(len(self.t_in_off_shifts[t, v, c]) for c in self.competencies if self.t_in_off_shifts.get((t,v,c))) * w[e, t, v]
+                <= quicksum(1 - y[c, e, t_mark] 
+                    for c in self.competencies
+                    if self.t_in_off_shifts.get((t,v,c))
+                    for t_mark in self.t_in_off_shifts[t, v, c]
```

Comment:
Same goes for this. Can maybe look over properly later today, but I do not have the time to figure it out myself right now.

Reply:
t_in_off_shift is now a set with indices t,v,c. A t is linked to a competency. It is therefore more efficient to only look at the t's connected to a certain competency.

Reply:
I just saw that t_in_off_shifts might not need to be connected to a competency. We could instead use an if-statement that checks if a "t" present in combined_time_periods have a demand for a certain competency

Reply:
I believe this method should work either way and might be more efficient so I'll let i stay for now

Reply:
I am happy if you are happy! Resolving this :)